### PR TITLE
Remember the maximization state of the window.

### PR DESCRIPTION
**Pull request by tsimonq2**
#450

### CODE REVIEW

**Comment by tsimonq2** [comment]
Note, this is somewhat inspired by [the PR adding this feature](https://github.com/lxqt/pcmanfm-qt/pull/124) in PCManFM-Qt.

**Comment by tsimonq2** [comment]
Tested this on Lubuntu Cosmic with Openbox, LXQt 0.13.0, and Qt 5.9.5. This feature works as intended.

File: src/main.cpp [hunk 1]
```diff
@@ -196,6 +196,8 @@ MainWindow *QTerminalApp::newWindow(bool dropMode, TerminalConfig &cfg)
     else
     {
         window = new MainWindow(cfg, dropMode);
+        if (Properties::Instance()->windowMaximized)
+            window->setWindowState(window->windowState() | Qt::WindowMaximized);
         window->show();
```
Comment (tsujan) [inline src/main.cpp]: `setWindowState` takes `Qt::WindowState` as its argument while you've given it `Qt::WindowStates`. Ref. http://doc.qt.io/qt-5/qwindow.html

Comment (tsimonq2) [reply src/main.cpp]: So, to make sure I get this right, the proper code looks like this?
```
window->setWindowStates(window->windowState() | Qt::WindowMaximized);
```

Comment (tsimonq2) [reply src/main.cpp]: To answer my own question, that would make sense, so I'll change it.

If I'm understanding this right, `Qt::WindowStates` can still take only one argument, so it wouldn't hurt either way to be safe here,

Comment (tsujan) [reply src/main.cpp]: My above comment may sound pedantic but future versions of Qt might not accept it. Please test `setWindowState(Qt::WindowMaximized);` it should be enough.

Comment (tsimonq2) [reply src/main.cpp]: Except, that's FTBFS...
```
src/main.cpp:200:21: error: ‘class MainWindow’ has no member named ‘setWindowStates’; did you mean ‘setWindowState’?
             window->setWindowStates(window->windowState() | Qt::WindowMaximized);
                     ^~~~~~~~~~~~~~~
                     setWindowState
```

Comment (tsimonq2) [reply src/main.cpp]: ACK

Comment (tsujan) [reply src/main.cpp]: My suggestion was `setWindowState(Qt::WindowMaximized)`.

**Comment by tsimonq2** [comment]
How's that look, @tsujan?

**Comment by tsujan** [comment]
It should be OK. Please test it too!

Quote from http://doc.qt.io/qt-5/qt.html#WindowState-enum: "The WindowStates type is a typedef for QFlags<WindowState>. It stores an OR combination of WindowState values."

**Comment by tsujan** [comment]
I should change it in pcmanfm-qt too.

**Comment by tsimonq2** [comment]
Just tested it with the same environment listed above, and it works fine.

Since this is a fairly simple patch, I'll merge it.

**Comment by tsujan** [comment]
GTM. Merge tit

**Comment by tsimonq2** [comment]
Thanks @tsujan! :smiley:

**Comment by tsujan** [comment]
You're welcome. My review was just about a tiny change.

**Pull request merged**